### PR TITLE
Use WASM bloom filter

### PR DESCRIPTION
**Pull request by PaulLeCam**
## Description

Replace the bloom filter library by the newly created https://github.com/ceramicnetwork/wasm-bloom-filter wrapping a Rust crate.

I used the [`growable-bloom-filter`](https://crates.io/crates/growable-bloom-filter) crate as it supports configuring the desired error probability and import/export to JSON, but I'm not familiar with the Rust ecosystem so if anyone has concerns or suggestions for better alternatives, please share them!

## How Has This Been Tested?

- [x] Updated existing tests

## Definition of Done

- [ ] The work addresses the description and outcomes in the issue
- [x] I have added relevant tests for new or updated functionality
- [x] My code follows conventions, is well commented, and easy to understand
- [x] My code builds and tests pass without any errors or warnings
- [x] I have tagged the relevant reviewers
- [ ] I have updated the READMEs of affected packages
- [ ] I have made corresponding changes to the documentation
- [x] The changes have been communicated to interested parties

### CODE REVIEW

File: src/merkle/merkle-objects.ts [hunk 1]
```diff
@@ -265,9 +268,14 @@ export class BloomMetadata implements MetadataFunction<Candidate, TreeMetadata>
...
-    const bloomFilter = BloomFilter.from(bloomFilterEntries, BLOOM_FILTER_FALSE_POSITIVE_RATE)
+
+    const bloomFilter = new BloomFilter(BLOOM_FILTER_FALSE_POSITIVE_RATE, bloomFilterEntries.size)
+    for (const entry of bloomFilterEntries) {
+      bloomFilter.add(entry)
+    }
+
     // @ts-ignore
-    const serializedBloomFilter = bloomFilter.saveAsJSON()
+    const serializedBloomFilter = bloomFilter.toString()
```
Comment (oed) [inline src/merkle/merkle-objects.ts:278]: Why is this a string and not a bytestring? Would be more efficient?

Comment (PaulLeCam) [reply src/merkle/merkle-objects.ts:278]: I followed the example from https://crates.io/crates/growable-bloom-filter to serialize to JSON, but yes I suppose we can implement a custom serialization if needed.

**Comment by oed** [comment]
@stbrody how do we feel about the parameterization of the lib?
https://github.com/ceramicnetwork/wasm-bloom-filter/blob/main/src/lib.rs#L31

File: src/merkle/merkle-objects.ts [hunk 2]
```diff
@@ -8,14 +8,17 @@ import { Request } from '../models/request.js'
...
 import { IpfsService } from '../services/ipfs-service.js'
 
-import { BloomFilter } from 'bloom-filters'
+import { BloomFilter } from '@ceramicnetwork/wasm-bloom-filter'
 import { StreamID } from '@ceramicnetwork/streamid'
 
 const packageJson = JSON.parse(
-  fs.readFileSync(new URL('../../node_modules/bloom-filters/package.json', import.meta.url), 'utf8')
+  fs.readFileSync(
+    new URL('../../node_modules/@ceramicnetwork/wasm-bloom-filter/package.json', import.meta.url),
```
[the remark this answers is not in the document]
Comment (PaulLeCam) [reply src/merkle/merkle-objects.ts:16]: Yes I think so, the WASM module is compiled locally before being published to npm, so I expected changes to the underlying library should only be reflected if we explicitly update it, compile and publish a new version.

File: src/merkle/merkle-objects.ts [hunk 3]
```diff
@@ -265,9 +268,14 @@ export class BloomMetadata implements MetadataFunction<Candidate, TreeMetadata>
...
       }
     }
-    const bloomFilter = BloomFilter.from(bloomFilterEntries, BLOOM_FILTER_FALSE_POSITIVE_RATE)
+
+    const bloomFilter = new BloomFilter(BLOOM_FILTER_FALSE_POSITIVE_RATE, bloomFilterEntries.size)
+    for (const entry of bloomFilterEntries) {
+      bloomFilter.add(entry)
+    }
+
     // @ts-ignore
```
[the remark this answers is not in the document]
Comment (PaulLeCam) [reply src/merkle/merkle-objects.ts]: Shouldn't be, I removed it 👍

**Comment by stbrody** [comment]
> @stbrody how do we feel about the parameterization of the lib? [ceramicnetwork/wasm-bloom-filter@`main`/src/lib.rs#L31](https://github.com/ceramicnetwork/wasm-bloom-filter/blob/main/src/lib.rs?rgh-link-date=2022-11-15T09%3A17%3A49Z#L31)

seems fine.  We're also not using the defaults, we initialize it with the known size of the filter at runtime: https://github.com/ceramicnetwork/ceramic-anchor-service/blob/chore/CDB-1962-wasm-bloom-filter/src/merkle/merkle-objects.ts#L272

**Comment by oed** [comment]
Great, should be good to merge then I guess!

**Pull request merged**